pkg/cluster/internal/loadbalancer/config.go

```diff
@@ -32,26 +32,33 @@ type ConfigData struct {
 
 // DefaultConfigTemplate is the loadbalancer config template
 const DefaultConfigTemplate = `# generated by kind
+global
+  log /dev/log local0
+  log /dev/log local1 notice
+  daemon
 
-# load balance over the control planes
-stream {
-    upstream tcp_backend {
-    {{- range $server, $address := .BackendServers}}
-        server {{ $address }};
-    {{- end}}
-    }
+defaults
+  log global
+  mode tcp
+  option dontlognull
+  # TODO: tune these
+  timeout connect 5000
+  timeout client 50000
+  timeout server 50000
 
-    server {
-        listen {{ .ControlPlanePort }};
-        {{ if .IPv6 -}}
-        listen [::]:{{ .ControlPlanePort }};
-        {{- end }}
-        proxy_pass tcp_backend;
-    }
-}
+frontend control-plane
+  bind *:{{ .ControlPlanePort }}
+  {{ if .IPv6 -}}
+  bind :::{{ .ControlPlanePort }};
+  {{- end }}
+  default_backend kube-apiservers
 
-# minimal events entry to make nginx happy
-events {}
+backend kube-apiservers
+  option httpchk GET /healthz
+  # TODO: we should be verifying (!)
```

Comment:
This aspect is super not ideal but could be solved later.
Ideally where possible kind should behave like production. There's no real reason we need to be more insecure here other than :effort: and logistics around the certs. We should be able to do SSL backends with verified certs.

Reply:
This is the status quo so it shouldn't block the PR, but ...

Reply:
this is an interesting topic. If we move that way we can use other load balancers, currently, we are load balancing at TCP level commonly named SSL Passtrough, and haproxy is the only one  (that I know) that allows to do health checking on a HTTPS endpoint.

If we are able to get the certificates to the load balancer we can start doing load balancer at L7